packages/flutter/lib/src/services/text_input.dart

```diff
@@ -671,6 +686,370 @@ class RawFloatingCursorPoint {
+        start: encoded['deltaStart'] as int? ?? -1,
+        end: encoded['deltaEnd'] as int? ?? -1,
+      ),
+      selection: TextSelection(
+        baseOffset: encoded['selectionBase'] as int? ?? -1,
+        extentOffset: encoded['selectionExtent'] as int? ?? -1,
+        affinity: _toTextAffinity(encoded['selectionAffinity'] as String?) ??
+            TextAffinity.downstream,
+        isDirectional: encoded['selectionIsDirectional'] as bool? ?? false,
+      ),
+      composing: TextRange(
+        start: encoded['composingBase'] as int? ?? -1,
+        end: encoded['composingExtent'] as int? ?? -1,
+      ),
+    );
+  }
+
+  /// {@macro flutter.services.TextEditingDelta.deltaType}
+  @override
+  TextEditingDeltaType get deltaType => TextEditingDeltaType.insertion;
+
+  /// {@macro flutter.services.TextEditingDelta.apply}
+  @override
+  TextEditingValue apply(TextEditingValue value) {
+    String newText = value.text;
+    newText = replace(newText, deltaText, deltaRange.start, deltaRange.end);
+    return value.copyWith(text: newText, selection: selection, composing: composing);
+  }
+}
+
+/// {@macro flutter.services.TextEditingDeltaDeletion}
+class TextEditingDeltaDeletion extends TextEditingDelta {
+  /// Creates a deletion delta for a given change to the editing state.
+  ///
+  /// {@macro flutter.services.TextEditingDelta}
+  const TextEditingDeltaDeletion({
+    required String oldText,
+    required String deltaText,
+    required TextRange deltaRange,
+    required TextSelection selection,
+    required TextRange composing,
+  }) : super(
+    oldText: oldText,
+    deltaText: deltaText,
+    deltaRange: deltaRange,
+    selection: selection,
+    composing:composing,
+  );
+
+  /// Creates an instance of this class from a JSON object.
+  factory TextEditingDeltaDeletion.fromJSON(Map<String, dynamic> encoded) {
+    return TextEditingDeltaDeletion(
+      oldText: encoded['oldText'] as String,
+      deltaText: encoded['deltaText'] as String,
+      deltaRange: TextRange(
+        start: encoded['deltaStart'] as int? ?? -1,
+        end: encoded['deltaEnd'] as int? ?? -1,
+      ),
+      selection: TextSelection(
+        baseOffset: encoded['selectionBase'] as int? ?? -1,
+        extentOffset: encoded['selectionExtent'] as int? ?? -1,
+        affinity: _toTextAffinity(encoded['selectionAffinity'] as String?) ??
+            TextAffinity.downstream,
+        isDirectional: encoded['selectionIsDirectional'] as bool? ?? false,
+      ),
+      composing: TextRange(
+        start: encoded['composingBase'] as int? ?? -1,
+        end: encoded['composingExtent'] as int? ?? -1,
+      ),
+    );
+  }
+
+  /// {@macro flutter.services.TextEditingDelta.deltaType}
+  @override
+  TextEditingDeltaType get deltaType => TextEditingDeltaType.deletion;
+
+  /// {@macro flutter.services.TextEditingDelta.apply}
+  @override
+  TextEditingValue apply(TextEditingValue value) {
+    String newText = value.text;
```

Comment:
Same as the previous comment. The base string should probably be `oldText`? Also how should conflicts be resolved if `value.text` is not `oldText`?